packages/master/master.js

```diff
@@ -259,6 +260,17 @@ function validateSlaveToken(req, res, next) {
 	next();
 }
 
+function findInstancesAssignedSlave(assigned_slave_id) {
```

Comment:
While on 5-10-20 slave instances this is fast and even 50 probably would be barely noticeable, if this is called in a lot of places (and other places in the code do the pattern), this will start to pile up as latency.
This should be a hash-map (associative array) lookup, not just looping over an array.

Reply:
This function is entirely redundant. `db.slaves` is a mapping of slave id to slave info objects. A call to `findInstancesAssignedSlave(...)` can be replaced with `db.slaves.get(...)`.

Reply:
i will be change to `db.slaves.get(...)`